includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1248,32 +1248,77 @@ public function save_intent_to_order( $order, $intent ) {
 	 * @return obect|bool     Either the intent object or `false`.
 	 */
 	public function get_intent_from_order( $order ) {
-		$order_id = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $order->id : $order->get_id();
-
 		if ( WC_Stripe_Helper::is_wc_lt( '3.0' ) ) {
-			$intent_id = get_post_meta( $order_id, '_stripe_intent_id', true );
-		} else {
-			$intent_id = $order->get_meta( '_stripe_intent_id' );
+			return $this->pre_3_0_get_intent_from_order( $order );
 		}
 
+		$order_id  = $order->get_id();
+		$intent_id = $order->get_meta( '_stripe_intent_id' );
+
 		if ( $intent_id ) {
-			return WC_Stripe_API::request( array(), "payment_intents/$intent_id", 'GET' );
+			return $this->get_intent( 'payment_intents', $intent_id );
 		}
 
 		// The order doesn't have a payment intent, but it may have a setup intent.
-		if ( WC_Stripe_Helper::is_wc_lt( '3.0' ) ) {
-			$intent_id = get_post_meta( $order_id, '_stripe_setup_intent', true );
-		} else {
-			$intent_id = $order->get_meta( '_stripe_setup_intent' );
+		$intent_id = $order->get_meta( '_stripe_setup_intent' );
+
+		if ( $intent_id ) {
+			return $this->get_intent( 'setup_intents', $intent_id );
 		}
 
+		return false;
+	}
+
+	/**
+	 * Retrieves the payment intent, associated with an order for WooCommerce < 3.0
+	 *
+	 * @param WC_Order $order The order to retrieve an intent for.
+	 * @return obect|bool     Either the intent object or `false`.
+	 */
+
+	private function pre_3_0_get_intent_from_order( $order ) {
```

Comment:
> But it might not need to be addressed since we'll be dropping support for WC 3.0 in the next release. Two functions might actually make it easier to remove the code.

That's exactly why I decided to extract pre 3.0 sequence into its own method 🙂
Do you still want me to put it back?